### ISSUE TITLE
hystrix-javanica not printing failure details in fallback

### ISSUE THREAD

**Issue by santoshavkr** [Netflix/Hystrix#1592]
We were using separate command classes for every hystrix command by extending  HystrixCommand . Then it was priting all the failures in fallback like example given below.

Execution events for *********Command THREAD_POOL_REJECTED

We migrated from separate command classes to annotation based command classes by using Javanica.

exception.getMessage()  which is mostly null in even if circuit breaks.

Is there a way to print the exact cause for why a command entered to fallback

**Comment by JagmohanSharma** [comment]
@santoshavkr  did you try using Throwable in fallback method to get the exact exception which caused command to fail?

**Comment by santoshavkr** [comment]
Yes Throwable.getMessage is not returning me any cause.It is mostly null except in case of any business failure by the calling service

**Comment by mattrjacobs** [comment]
@dmgcodevil Can you see if this is working expected in Javanica?

**Comment by dmgcodevil** [comment]
@mattrjacobs will take a look shortly

**Comment by dmgcodevil** [comment]
@santoshavkr which version do you use ?

**Comment by bjoern-dev** [comment]
Hi, thanks for the awesome work with javanica.
We are having a very similar problem here: we'd like to migrate to javanica. However, we got stuck at the point where we need to log detailled information within the fallback method, like command.getMetrics().getCurrentConcurrentExecutionCount() etc.

We need a way to get a hold of the current HystrixCommand Object.
Do you think it would be possible and a viable idea to add a third optional parameter to the fallback method, that holds the HystrixCommand object?

**Comment by dmgcodevil** [comment]
Hi @bjoern-dev third parameter sounds tricky, I wonder if Hystrix already provides some way to get currently executing command, is there anything like `HystrixContext.getCurrentCommand()`, @mattrjacobs  ?

**Comment by mattrjacobs** [comment]
@dmgcodevil @bjoern-dev I'm no longer working at Netflix, and no longer working on Hystrix.  Someone else from Netflix should hopefully be around to help move things along.  Thanks!
